src/seq_join.rs

```diff
@@ -117,14 +117,55 @@ where
     seq_join(active, iterable).try_collect()
 }
 
+enum ActiveItem<F: IntoFuture> {
+    Pending(Pin<Box<F::IntoFuture>>),
+    Resolved(F::Output),
+    Done,
+}
+
+impl<F: IntoFuture> ActiveItem<F> {
+    /// Drives this item to resolved state when value is ready to be taken out. Has no effect
+    /// if the value is ready.
+    ///
+    /// ## Panics
+    /// Panics if polled after completion.
+    fn poll(&mut self, cx: &mut Context<'_>) -> Poll<()> {
```

Comment:
yea `#[allow(unused_must_use)]` is ugly. I still like `Poll` interface here because it better represents the status of this operation, but `#[must_use]` on `Poll` kills it. 

Changed it to return `bool` instead.